contract/src/msgs/sorted_set.rs

```diff
@@ -0,0 +1,99 @@
+use seda_common::msgs::data_requests::DataRequest;
+
+use super::*;
+
+pub type IndexKey = (u128, u64, Hash);
+
+pub struct SortedSet<'a> {
+    #[cfg(test)]
+    pub len:            Item<u32>,
+    pub index:          Map<IndexKey, ()>,
+    pub dr_id_to_index: Map<&'a Hash, IndexKey>,
+}
+
+impl SortedSet<'_> {
+    #[cfg(test)]
+    pub fn initialize(&self, store: &mut dyn Storage) -> StdResult<()> {
+        self.len.save(store, &0)?;
+        Ok(())
+    }
+
+    #[cfg(test)]
+    pub fn len(&self, store: &dyn Storage) -> StdResult<u32> {
+        self.len.load(store)
+    }
```

Comment:
why only for test?
I suppose because we are not using it apart for tests...
could it be interesting for pagination purposes when querying requests by status.

Reply:
Even if we don't use it only for tests it won't have any effect on pagination. No way for it to.

Reply:
I meant, that maybe we could also return the length when querying items, because it that way solvers could have more information about how many more requests are remaining.

Reply:
As discussed it's total instead of remaining.